docs/ref/checks.txt

```diff
@@ -330,6 +330,8 @@ Models
   ``<max_length>`` characters.
 * **models.W035**: ``db_table`` ``<db_table>`` is used by multiple models:
   ``<model list>``.
+* **models.W036**: ``<database>`` does not support partial indexing, so the
+  unique constraint won't be created.
```

Comment:
If the message shown is updated as I suggested, this should probably follow that wording.